src/Sarif/IResultLogWriter.cs

```diff
@@ -41,15 +41,15 @@ public interface IResultLogWriter
         void WriteLogicalLocations(IList<LogicalLocation> logicalLocations);
 
         /// <summary>
-        /// Write information about rules to the log. This information may appear
-        /// after the results, as the relevant set of rules might not be known until
-        /// all results have been generated. A Sarif file may also contain only rules
-        /// metadata.
+        /// Write information about the tool to the log. This information may appear
+        /// after the results, as some tool data, such as the relevant set of rule
+        /// observed during analysis, might not be known until all results have been
+        /// generated. A Sarif file may also contain only message metadata.
         /// </summary>
-        /// <param name="rules">
-        /// An array whose elements comprise rules relevant to the analysis.
+        /// <param name="tool">
+        /// The tool object relevant to the current run.
         /// </param>
-        void WriteRules(IList<Rule> rules);
+        void WriteTool(Tool tool);
```

Comment:
>WriteTool [](start = 13, length = 9)

In general, the SARIF design has changed sufficiently that the approach of this logger isn't ideal (it wasn't a great design to begin with). Moving forward, I think this logger could usefully work as follow: initialize the logger with a run object, user then writes results. logger accumulates file objects, etc., to the run object associated with the log. On Dispose, the log would write all non-null run objects, for logical locations, etc. Consumer could still update these run objects on the fly if useful. By using this approach, the logger interface could be greatly reduced (i.e., no need to maintain helpers to write all the various tables).

Reply:
I like that idea.

---
In reply to: [255236265](https://github.com/Microsoft/sarif-sdk/pull/1259#discussion_r255236265) [](ancestors = 255236265)